File: 3541-find-most-frequent-vowel-and-consonant/3541-find-most-frequent-vowel-and-consonant.java
class Solution {
    private boolean isVowel(int c) {
        return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u';
    }
    public int maxFreqSum(String s) {
         return s.chars()
                .boxed()
                .collect(Collectors.groupingBy(c -> !isVowel(c),Collectors.groupingBy(c -> c, Collectors.counting())))
                .values().stream()
                .mapToInt(m -> m.values().stream().mapToInt(Long::intValue).max().orElse(0))
                .sum();
    }
}

File: 3541-find-most-frequent-vowel-and-consonant/3541-find-most-frequent-vowel-and-consonant.cpp
class Solution {
private:
    bool is_vowel(char c){
        return unordered_set<char>{'a', 'e', 'i', 'o', 'u'}.count(tolower(c));
    }
public:
    int maxFreqSum(string s) {
        vector<int>map(26,0);
        int max_v = 0, max_c = 0;
        for(auto const &c : s){
            map[c-'a']++;
            if(is_vowel(c))
                max_v = max(max_v , map[c-'a']);
            else
                max_c = max(max_c, map[c-'a']);    
        }
        return max_v + max_c;
    }
};